### PR TITLE
Removes scala actors from eclipse repl and scala ide in general

### PR DESCRIPTION
**Pull request by wpopielarski**
Because scala-actors can be sunset in 2.12 this PR replaces actors with scala `Future`s

### CODE REVIEW

**Comment by sschaef** [comment]
I just want to point it out as a TODO point: The Scala actors dependency should be removed from our build, this way we can be sure that it is really gone.

File: org.scala-ide.sdt.core.tests/pom.xml [hunk 1]
```diff
@@ -14,6 +14,30 @@
...
+        </exclusion>
+        <exclusion>
+          <groupId>com.sun.jdmk</groupId>
+          <artifactId>jmxtools</artifactId>
+        </exclusion>
+        <exclusion>
+          <groupId>com.sun.jmx</groupId>
+          <artifactId>jmxri</artifactId>
+        </exclusion>
+      </exclusions>
```
Comment (sschaef) [inline org.scala-ide.sdt.core.tests/pom.xml]: Why do we need this exclusions?

Comment (wpopielarski) [inline org.scala-ide.sdt.core.tests/pom.xml]: I don't know, they are just moved and have been existing before my change

2016-01-05 14:13 GMT+01:00 Simon Schäfer <notifications@github.com>:

**Comment by sschaef** [comment]
Code looks ok, I didn't test it yet though. Also, I found still some references to `actor` in the codebase, mostly in comments.

```
% ag "\bactor\b"
org.scala-ide.sdt.debug/src/org/scalaide/debug/internal/preferences/AsyncDebuggerPreferencePage.scala
135:      "akka.actor.LocalActorRef,$bang,0",
136:      "akka.actor.RepointableActorRef,$bang,0",

org.scala-ide.sdt.debug/src/org/scalaide/debug/internal/breakpoints/BreakpointSupport.scala
132:  /** Register `this` actor to receive all notifications from the `eventDispatcher` related to the passed `request`.*/

org.scala-ide.sdt.debug/src/org/scalaide/debug/internal/breakpoints/ScalaDebugBreakpointManager.scala
39: *       actor. This seems useless (listeners are run in their own thread) and makes things somewhat harder to test.
40: *       Maybe we should remove the companion actor in this case.

org.scala-ide.sdt.debug/src/org/scalaide/debug/internal/model/ScalaJdiEventDispatcher.scala
87:   * Register the actor as recipient of the call back for the given request
88:   * TODO: I think we should try to use JDI's mechanisms to associate the actor to the request:
132:   * Resume or not the stopped threads depending on actor's answers
146:    // forward each event to the interested actor
151:        /* TODO: I think we should try to use JDI's mechanisms to associate the actor to the request:

org.scala-ide.sdt.debug/src/org/scalaide/debug/internal/model/ScalaThread.scala
243:   * Methods used by the companion actor to update this object internal states

org.scala-ide.sdt.debug/src/org/scalaide/debug/internal/model/ScalaDebugTarget.scala
342:   * Called when attaching to a remote VM. Makes the companion actor run the initialization
345:   *  @note This method has no effect if the actor was already initialized
351:   * Methods used by the companion actor to update this object internal states
356:   * Callback form the actor when the connection with the vm is enabled.
375:  /** Callback from the actor when the connection with the vm as been lost

org.scala-ide.sdt.debug/src/org/scalaide/debug/internal/model/ScalaLogicalStructureProvider.scala
166:    ScalaDebugPlugin.id + "logicalstructure.actor"
170:      val Some(actor) = enclosingActor(ref)
171:      implicit val target = actor.getDebugTarget()
172:      val actorContext = actor.invokeMethod("context", ScalaDebugger.currentThread).asInstanceOf[ScalaObjectReference]
173:      val sender = actor.invokeMethod("sender", ScalaDebugger.currentThread)
174:      val parent = actorContext.invokeMethod("parent", "()Lakka/actor/ActorRef;", ScalaDebugger.currentThread)
175:      val supervisingStrategy = actor.invokeMethod("supervisorStrategy", "()Lakka/actor/SupervisorStrategy;", ScalaDebugger.cur
rentThread)
177:      VirtualValue("Actor", actor.invokeMethod("self", ScalaDebugger.currentThread).getValueString())
189:    obj.wrapJDIException("Exception while computing logical structures for actor") {
191:        if (implements(obj.classType, "akka.actor.Actor")) Some(obj)

org.scala-ide.sdt.debug/src/org/scalaide/debug/internal/model/ScalaDebugCache.scala
62: * Most of the methods are synchronous calls made to the underlying actor.
79:        logger.info("TIMEOUT waiting for debug cache actor in getLoadedNestedTypes")
285:        }) logger.info("TIMOUT waiting for the listener actor in `classLoaded`")

org.scala-ide.sdt.debug/src/org/scalaide/debug/internal/hcr/ScalaHotCodeReplaceManager.scala
46: * Monitors changed resources and notifies debug target's companion actor, when there are some changed class files

org.scala-ide.sdt.debug/src/org/scalaide/debug/internal/ScalaDebugPlugin.scala
22:  final val IMG_ACTOR = "images.actor"
51:    declareImage(ScalaDebugPlugin.IMG_ACTOR, "icons/actor-2.png")

org.scala-ide.sdt.debug/src/org/scalaide/debug/internal/async/BreakOnDeadLetters.scala
31:  private val asyncPoint = AsyncProgramPoint("akka.actor.DeadLetterActorRef", "$bang", 0)

org.scala-ide.sdt.debug/src/org/scalaide/debug/internal/async/StepMessageOut.scala
41:    AsyncProgramPoint("akka.actor.RepointableActorRef", "$bang", 0),
42:    AsyncProgramPoint("akka.actor.LocalActorRef", "$bang", 0),
45:    AsyncProgramPoint("akka.actor.ActorCell", "receiveMessage", 0))

org.scala-ide.sdt.debug.tests/src/org/scalaide/debug/internal/model/ScalaDebuggerCacheTest.scala
85:    // a latch listener actor. It releases the latch when it receives a ClassPrepareEvent
109:    assertTrue("Message was not received by listening actor before timeout", testing.awaitLatch(500))
118:    assertFalse("Unexpected message was received by listening actor before timeout", testing.awaitLatch(500))

org.scala-ide.sdt.debug.tests/src/org/scalaide/debug/internal/model/ScalaThreadTest.scala
109:   * Check that calling #getStackFrame doesn't create a freeze. It used to be making a sync call to the actor, even if it was s
hutdown.

org.scala-ide.sdt.debug.tests/src/org/scalaide/debug/internal/model/ScalaDebugTargetTest.scala
93:   * Check that calling #getThreads doesn't create a freeze. It used to be making a sync call to the actor, even if it was shutd
own.

org.scala-ide.sdt.core/templates/default-templates.xml
3:<template name="actor"
4:  description="actor statement"
5:  id="scala.templates.actor"
7:>val a = actor {

org.scala-ide.sdt.core/src/org/scalaide/util/Utils.scala
58:   * Scala wrapper on [[JdiInvocationSynchronizer]]. Remark: the actor thread can be killed in any moment
```

**Comment by wpopielarski** [comment]
references to akka.actor are really ok, others need to be updated, will work on it in separate commit (changes in comment and potentially text formatting)

**Comment by wpopielarski** [comment]
retest this please

**Comment by sschaef** [comment]
One remark left, beside from that LGTM.

**Pull request merged**